src/python/pants/backend/tools/yamllint/rules.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+import os
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Any, cast
+
+from pants.backend.python.util_rules.pex import Pex, PexProcess, PexRequest
+from pants.backend.tools.yamllint.subsystem import Yamllint
+from pants.core.goals.lint import LintFilesRequest, LintResult
+from pants.core.goals.tailor import group_by_dir
```

Comment:
It's only a little weird we're importing this from `tailor`...

We should probably put this in a utility file :smile:

Reply:
Seems like a good place to me :+1:

Reply:
I've made a PR for that at #18029.